### ISSUE TITLE
Passing a fluid object first and then a background stack (array) causes crash

### ISSUE THREAD

**Issue by free-easy** [timhagn/gatsby-background-image#84]
## Description

Passing a fluid object first and then a background stack (array) leads to error

### Steps to reproduce

1. Pass a fluid object in the fluid prop
2. Update the state by passing a background stack array (i.e. `(IFluidObject | string)[]`) to the fluid prop
3. Error

### Expected result

The new background image and linear-gradient should be set

### Actual result

[localhost-1572429439618.log](https://github.com/timhagn/gatsby-background-image/files/3788107/localhost-1572429439618.log)

### Environment

```
  System:
    OS: Linux 4.15 Ubuntu 18.04.3 LTS (Bionic Beaver)
    CPU: (4) x64 Intel(R) Core(TM) i5-5300U CPU @ 2.30GHz
    Shell: 4.4.20 - /bin/bash
  Binaries:
    Node: 10.16.3 - /usr/bin/node
    Yarn: 1.19.1 - ~/.npm/bin/yarn
    npm: 6.10.1 - ~/.npm/bin/npm
  Languages:
    Python: 3.6.8 - /usr/bin/python
  Browsers:
    Firefox: 69.0.2
  npmGlobalPackages:
    gatsby-cli: 2.8.8
```

**Comment by timhagn** [comment]
Hi @free-easy,

thanks for noticing & the log, didn't happen to me up till now. Gonna try squeezing this in on Friday / the weekend, but sadly no promises atm % ).

Best,

Tim.

**Comment by free-easy** [comment]
Thank you for the quick response.
Let me know if there's anything I can help with.

**Comment by timhagn** [comment]
Oh, I'd be all to happy for any PR at the moment : ).

**Comment by free-easy** [comment]
I'll try to do my best xD.

**Comment by timhagn** [comment]
Hi @free-easy,

the behavior should be rectified for the moment with the just published `v0.8.16` : ).
Would you be so kind to test it on your side for any unwanted side-effects?

Best,

Tim.

**Comment by free-easy** [comment]
@timhagn Thank you very much for your quick fix! I confirm that everything works!
